### PR TITLE
Improvements in ac-stroop

### PR DESCRIPTION
**Pull request by lfaucon**
@houshuang You merged my previous PR quite prematurily (CI was red and it had the "not ready" tag). This version should be much better. Let's try it out Tomorrow with Jenny also to discuss the last things we need to implement.

### CODE REVIEW

File: ac/ac-stroop/src/ActivityRunner.jsx [hunk 1]
```diff
@@ -42,80 +49,136 @@ const generateExample = (objects, colors) => {
...
-  const objectList = objects.split(',')
+let noAnswerTimeout;
 
-  const nextQuestion = () => {
-    setQuestion(generateExample(objectList, colorList))
-  }
+const Question = props => {
+  const { setQuestion, question, progress, setProgress } = props;
+  const { objectName, colorName, colorFill } = question;
+  const { logger } = props;
```
Comment (houshuang) [inline ac/ac-stroop/src/ActivityRunner.jsx]: why not on line 98? :)

File: ac/ac-stroop/src/ActivityRunner.jsx [hunk 2]
```diff
@@ -42,80 +49,136 @@ const generateExample = (objects, colors) => {
...
+    logger({ type: 'progress', value: progress + 1 });
+    setProgress(progress + 1);
+    setQuestion('waiting');
+  };
+
+  Mousetrap.bind('y', onClick(true));
+  Mousetrap.bind('n', onClick(false));
+
+  clearTimeout(noAnswerTimeout);
+  noAnswerTimeout = setTimeout(onClick(undefined), 10000);
```
Comment (houshuang) [inline ac/ac-stroop/src/ActivityRunner.jsx:116]: Shouldn't this be setTimeout(() => onClick(undefined))?

Comment (houshuang) [reply ac/ac-stroop/src/ActivityRunner.jsx:116]: Ah ok, onClick returns a function. C'est tout bon :)

File: ac/ac-stroop/src/Dashboard.jsx [hunk 3]
```diff
@@ -1,29 +1,35 @@
...
+      answerTime
+    } = log.payload;
+    const qType = isConsistent ? 'consistent' : 'inconsistent';
+    if (isCorrect === answer) {
+      dataFn.numIncr(1, [qType, 'correct', 'count']);
+      dataFn.numIncr(answerTime - startTime, [qType, 'correct', 'time']);
     } else {
-      dataFn.numIncr(1, [stroopType, 'wrong', 'count'])
+      dataFn.numIncr(1, [qType, 'wrong', 'count']);
+      dataFn.numIncr(answerTime - startTime, [qType, 'wrong', 'time']);
```
Comment (houshuang) [inline ac/ac-stroop/src/Dashboard.jsx]: I don't understand this - you only store the total wrong time and total right time for all questions? Is this what the dashboard needs?

**Review by houshuang — CHANGES_REQUESTED** [review]

**Comment by houshuang** [comment]
You're not storing current state in reactive data, which means that on reload you restart the exercise?

File: ac/ac-stroop/src/Dashboard.jsx (hunk 3 above)
Comment (lfaucon) [reply ac/ac-stroop/src/Dashboard.jsx]: I store the sum and the count of all the time spent on questions for each of the 4 category (inconsitent/consistent x correct/wrong). That allows to compute the averages we need for the dashboard.

Comment (jkolsen) [reply ac/ac-stroop/src/Dashboard.jsx]: In the current dashboards, they are not distinguishing the correct/wrong but just has graphs for the inconsistent/consistent and checking to see if the student correctly marked it as wrong or not.

**Comment by lfaucon** [comment]
Should be good to go now

**Pull request merged**